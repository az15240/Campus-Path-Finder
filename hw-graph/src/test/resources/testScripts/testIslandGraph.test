# Test a graph with islands. Island means having discontinuous pieces of sub-graphs.
# Graph ('A') and graph ('B') all have two islands.

CreateGraph A
# Create the first island
AddNode A n1
AddNode A n2
AddNode A n3
AddEdge A n1 n2 e12
AddEdge A n2 n3 e23
AddEdge A n3 n1 e31
# Create the second island - only a single node
AddNode A n4
ListNodes A
ListChildren A n1
ListChildren A n2
ListChildren A n3
ListChildren A n4

CreateGraph B
# Create the first island
AddNode B n1
AddNode B n2
AddNode B n3
AddEdge B n1 n2 e12
AddEdge B n2 n3 e23
AddEdge B n3 n1 e31
# Create the second island
AddNode B n4
AddNode B n5
AddNode B n6
AddEdge B n4 n5 e45
AddEdge B n5 n6 e56
ListNodes B
ListChildren B n1
ListChildren B n2
ListChildren B n3
ListChildren B n4
ListChildren B n5
ListChildren B n6