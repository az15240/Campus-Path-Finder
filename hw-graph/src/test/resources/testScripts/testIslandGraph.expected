# Test a graph with islands. Island means having discontinuous pieces of sub-graphs.
# Graph ('A') and graph ('B') all have two islands.

created graph A
# Create the first island
added node n1 to A
added node n2 to A
added node n3 to A
added edge e12 from n1 to n2 in A
added edge e23 from n2 to n3 in A
added edge e31 from n3 to n1 in A
# Create the second island - only a single node
added node n4 to A
A contains: n1 n2 n3 n4
the children of n1 in A are: n2(e12)
the children of n2 in A are: n3(e23)
the children of n3 in A are: n1(e31)
the children of n4 in A are:

created graph B
# Create the first island
added node n1 to B
added node n2 to B
added node n3 to B
added edge e12 from n1 to n2 in B
added edge e23 from n2 to n3 in B
added edge e31 from n3 to n1 in B
# Create the second island
added node n4 to B
added node n5 to B
added node n6 to B
added edge e45 from n4 to n5 in B
added edge e56 from n5 to n6 in B
B contains: n1 n2 n3 n4 n5 n6
the children of n1 in B are: n2(e12)
the children of n2 in B are: n3(e23)
the children of n3 in B are: n1(e31)
the children of n4 in B are: n5(e45)
the children of n5 in B are: n6(e56)
the children of n6 in B are: